code/Controllers/CMSPageHistoryController.php

```diff
@@ -475,4 +472,19 @@ public function Breadcrumbs($unlinked = false)
         $crumbs[0]->Title = _t('CMSPagesController.MENUTITLE', 'Pages');
         return $crumbs;
     }
+    
+    public function transformReadonly(FieldList $fields)
+    {
+        foreach($fields as &$field) {
```

Comment:
The problem with iterating over FieldList directly is that it only loops over the top level of fields.

Try `foreach ($fields->dataFields() as $field) {`, which has the added benefit of selecting only data fields, so you don't need the CompositeField filter.